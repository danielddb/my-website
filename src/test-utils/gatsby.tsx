import { FixedObject } from 'gatsby-image';

export const createGatsbyFixedImageMock = (): FixedObject => ({
  base64:
    'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAUABQDASIAAhEBAxEB/8QAGAABAAMBAAAAAAAAAAAAAAAAAAECBAX/xAAXAQEBAQEAAAAAAAAAAAAAAAABAwAC/9oADAMBAAIQAxAAAAHPVWbVIOrmLbUHn//EABwQAAICAgMAAAAAAAAAAAAAAAABAhEDEBIhMf/aAAgBAQABBQKy6b94tbydmGCHBM//xAAUEQEAAAAAAAAAAAAAAAAAAAAg/9oACAEDAQE/AR//xAAVEQEBAAAAAAAAAAAAAAAAAAABEP/aAAgBAgEBPwEjP//EABoQAAIDAQEAAAAAAAAAAAAAAAARARAxIVL/2gAIAQEABj8CUSj0cZlYK//EABwQAAIDAAMBAAAAAAAAAAAAAAABESExUWFxkf/aAAgBAQABPyHkexBVwy2JbD0c5vD+HUIh8hFdg6m2pgts8P/aAAwDAQACAAMAAAAQrAeA/8QAGBEAAwEBAAAAAAAAAAAAAAAAAAExEBH/2gAIAQMBAT8QqOijz//EABcRAQEBAQAAAAAAAAAAAAAAAAEAETH/2gAIAQIBAT8QOFjdEht//8QAHhABAQACAgIDAAAAAAAAAAAAAREAITFBUWFxgaH/2gAIAQEAAT8QlUAsjTlUNNinPd8mSN7O+H5hwCbq2jEs+VNXA+J9QcoDUX7856WQ8YuQ6yOGf//Z',
  width: 64,
  height: 64,
  src: '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg',
  srcSet:
    '/static/6f6ccd0263b69cfd693f71c17be8ac7c/cf62c/avatar.jpg 1x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/eee8e/avatar.jpg 1.5x,\n/static/6f6ccd0263b69cfd693f71c17be8ac7c/6dc0c/avatar.jpg 2x'
});
