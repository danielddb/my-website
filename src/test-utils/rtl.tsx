import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import light from '../themes/light';

const Wrapper = ({ children }) => (
  <ThemeProvider theme={light}>{children}</ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: Wrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
