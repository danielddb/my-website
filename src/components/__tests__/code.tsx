import { axe } from 'jest-axe';
import React from 'react';
import { render as rtlRender } from 'test-utils/rtl';
import Code from '../code';

function render() {
  const codeString = `
  const routes: Routes = [
    {
      path: 'customers',
      loadChildren: () => import('./customers/customers.module').then(m => m.CustomersModule)
    }
  ];
  `.trim();

  return {
    totalLines: 6,
    ...rtlRender(<Code codeString={codeString} language={`ts`} />)
  };
}

test('it should render correctly', () => {
  const { container } = render();
  expect(container).toMatchSnapshot();
});

test('it should render the line numbers correctly', () => {
  const { totalLines, getAllByTestId } = render();
  const lineNumbers = getAllByTestId('line-number').map(node =>
    node.textContent.trim()
  );
  expect(lineNumbers).toHaveLength(totalLines);
  expect(lineNumbers).toMatchInlineSnapshot(`
    Array [
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
    ]
  `);
});

test('it should render with no accessibility issues', async () => {
  const { container } = render();
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
