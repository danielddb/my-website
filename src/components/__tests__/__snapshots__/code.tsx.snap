// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render correctly 1`] = `
.c0 {
  position: relative;
}

.c1 {
  text-align: left;
  padding: 0.5rem;
  overflow-x: auto;
  border-radius: 0.25rem;
}

.c2 {
  display: inline-block;
  width: 2em;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  opacity: 0.3;
}

<div>
  <div
    class="c0"
  >
    <pre
      class="c1 prism-code language-ts"
      style="color: rgb(214, 222, 235); background-color: rgb(1, 22, 39);"
    >
      <div
        class="token-line"
        style="color: rgb(214, 222, 235);"
      >
        <span
          class="c2"
          data-testid="line-number"
        >
          1
        </span>
        <span
          class="token keyword"
          style="color: rgb(127, 219, 202);"
        >
          const
        </span>
        <span
          class="token plain"
        >
           routes
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          :
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token maybe-class-name"
        >
          Routes
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token operator"
          style="color: rgb(127, 219, 202);"
        >
          =
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          [
        </span>
        <span
          class="token plain"
        />
      </div>
      <div
        class="token-line"
        style="color: rgb(214, 222, 235);"
      >
        <span
          class="c2"
          data-testid="line-number"
        >
          2
        </span>
        <span
          class="token plain"
        >
              
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          {
        </span>
        <span
          class="token plain"
        />
      </div>
      <div
        class="token-line"
        style="color: rgb(214, 222, 235);"
      >
        <span
          class="c2"
          data-testid="line-number"
        >
          3
        </span>
        <span
          class="token plain"
        >
                path
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          :
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token string"
          style="color: rgb(173, 219, 103);"
        >
          'customers'
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          ,
        </span>
        <span
          class="token plain"
        />
      </div>
      <div
        class="token-line"
        style="color: rgb(214, 222, 235);"
      >
        <span
          class="c2"
          data-testid="line-number"
        >
          4
        </span>
        <span
          class="token plain"
        >
                
        </span>
        <span
          class="token function-variable function"
          style="color: rgb(130, 170, 255);"
        >
          loadChildren
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          :
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          (
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          )
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token arrow operator"
          style="color: rgb(127, 219, 202);"
        >
          =&gt;
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token keyword"
          style="color: rgb(127, 219, 202);"
        >
          import
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          (
        </span>
        <span
          class="token string"
          style="color: rgb(173, 219, 103);"
        >
          './customers/customers.module'
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          )
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          .
        </span>
        <span
          class="token method function property-access"
          style="color: rgb(130, 170, 255);"
        >
          then
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          (
        </span>
        <span
          class="token parameter"
        >
          m
        </span>
        <span
          class="token plain"
        >
           
        </span>
        <span
          class="token arrow operator"
          style="color: rgb(127, 219, 202);"
        >
          =&gt;
        </span>
        <span
          class="token plain"
        >
           m
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          .
        </span>
        <span
          class="token property-access maybe-class-name"
        >
          CustomersModule
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          )
        </span>
        <span
          class="token plain"
        />
      </div>
      <div
        class="token-line"
        style="color: rgb(214, 222, 235);"
      >
        <span
          class="c2"
          data-testid="line-number"
        >
          5
        </span>
        <span
          class="token plain"
        >
              
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          }
        </span>
        <span
          class="token plain"
        />
      </div>
      <div
        class="token-line"
        style="color: rgb(214, 222, 235);"
      >
        <span
          class="c2"
          data-testid="line-number"
        >
          6
        </span>
        <span
          class="token plain"
        >
            
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          ]
        </span>
        <span
          class="token punctuation"
          style="color: rgb(199, 146, 234);"
        >
          ;
        </span>
      </div>
    </pre>
  </div>
</div>
`;
